equipment/equipment.lua

```diff
@@ -9,7 +10,70 @@ ftype.base.fields.item.type.fields.item = {
     end,
 }
 
-return data
+ftype.base.fields.equip = {
+    data = function(equipment_slot, item)
+        local bag = item.bag
+        local index = item.index
+        assert(resources.bags[bag].equippable, "Can not equip from this bag (bag = " .. bag .. ")")
+        assert(item.id ~= 0, "Can not equip from an empty bag slot (bag = " .. bag .. ", index = " .. index .. ")")
+
+        packets.outgoing[0x050]:inject({bag_index = index, slot_id = equipment_slot.slot, bag_id = bag})
+    end,
+}
+
+ftype.base.fields.unequip = {
+    data = function(equipment_slot)
+        packets.outgoing[0x050]:inject({bag_index = 0, slot_id = equipment_slot.slot, bag_id = 0})
+    end,
+}
+
+local equipment = {}
+
+equipment.equip = function(slot_items)
+    local count = 0
+    local items = {}
+    for slot, item in pairs(slot_items) do
+        if slot >= 0 and slot <= 15 then
+            local bag = item and item.bag or 0
+            local index = item and item.index or 0
+            assert(resources.bags[bag].equippable, "Can not equip from this bag (bag = " .. bag .. ")")
+            assert(not item or item.id ~= 0, "Can not equip from an empty bag slot (bag = " .. bag .. ", index = " .. index .. ")")
```

Comment:
You can actually pass `false` instead of an `item_type` object. I did this to match the functionality of `/equipset` in the game where each slot can be equipped, unequipped or not touched. So if you pass `false` for an equipment slot it will unequip that slot. And that's why lines 37 and 38 are needed.

I can do it a different way if you prefer, but I feel like it's pretty important to have that ability in some way.

Reply:
That is a good point that I had not considered. Though I would check `item == false` in that case to make it more obvious.

Thinking about it though, we do have an empty item type, which already has those fields set to `0`. We could let users use that instead of `false`, I feel that would make for a neater API:
```lua
equipment:equip({
    [0] = items.bags[0][12],
    [1] = items.bags[0][37],
    [9] = items.empty,
    [10] = items.bags[8][44],
})
```

I'm liking this idea, but the `items_service` does not expose its empty item yet. This could be changed very easily though. How do you feel about this? The entire checking code (and the function's code in general) would be reduced very heavily. Could also remove the slot index checks by changing the loop itself to `for i = 0, 15 do`. It would also be faster and process the set in order, though that likely won't matter much, but if you get determinism for free, I'd take it :D

Reply:
Yeah, I like this plan.

Reply:
Made those changes but I still need to differentiate between an empty bag slot and `items.empty` (also I think `items.none` might be a better name?) so it doesn't really reduce the code that much. But I still like it better.